### PR TITLE
Merges HEADBANGPROTECT and EARBANGPROTECT into new secondary flag

### PR DESCRIPTION
**Pull request by coiax**
Unless we get a clothing item that can be worn on the head, and the
ears, and only protects from loud noises in one of the configurations,
this isn't going to be an issue.

Frees up another flag slot, all hail the secondary flags.

### CODE REVIEW

File: code/__DEFINES/flags.dm [hunk 1]
```diff
@@ -19,11 +19,7 @@
 #define OVERLAY_QUEUED  256		//atom queued to SSoverlay
 #define ON_BORDER		512		// item has priority to check when entering or leaving
 
-#define EARBANGPROTECT		1024
-
-#define NOSLIP		1024 		//prevents from slipping on wet floors, in space etc (NOTE: flag shared with THICKMATERIAL for external suits and helmet)
-
-#define HEADBANGPROTECT		4096
+#define NOSLIP		1024 		//prevents from slipping on wet floors, in space etc
```
Comment (coiax) [inline code/__DEFINES/flags.dm]: It no longer shares a flag with THICKMATERIAL.

Comment (RemieRichards) [inline code/__DEFINES/flags.dm]: give its slot to BLOCK_GAS_SMOKE_EFFECT, since it current has to share with THICKMATERIAL

Comment (coiax) [inline code/__DEFINES/flags.dm]: But it doesn't share with THICKMATERIAL. THICKMATERIAL is 8192, which it shares with BLOCK_GAS_SMOKE_EFFECT.

Comment (RemieRichards) [inline code/__DEFINES/flags.dm]: please read what I said?? (the edit, if you're paranoid, is current->currently, check your emails if you get them to confirm)

Comment (coiax) [inline code/__DEFINES/flags.dm]: But the slots are shared when there's no risk of them clashing. I assume BLOCK_GAS is only for masks, and THICK is for exo and undersuits.

Comment (RemieRichards) [inline code/__DEFINES/flags.dm]: I disagree with the sharing, at all, we should just have separate vars as far as I'm concerned.

Comment (MrPerson) [inline code/__DEFINES/flags.dm]: I agree with remie, sharing is stupid.

Comment (coiax) [inline code/__DEFINES/flags.dm]: That's for another PR then, all I did was remove the misleading comment.

**Pull request merged**